functions/src/modules/SE.functions.ts

```diff
@@ -77,3 +79,94 @@ export const processNewAllotment = functions.database
     });
     return 1;
 });
+
+
+
+
+/////////////////////////////////////////////////
+//
+//          UploadProcessing (Storage Trigger)
+//      1. Send a notification to the coordinator
+//      2. Update the Task
+//
+/////////////////////////////////////////////////
+
+export const uploadProcessing = functions.storage.object()
+.onFinalize(async object => {    
+
+    const filePath = object.name;
+
+
+    if (!filePath.startsWith('sound-editing/restored/uploads'))
+        return -1;
+    
+    
+    const filePathRegex = /sound-editing\/restored\/uploads\/(\w+)\/(\w+)\/(\w+\/)*(([\w\d]+)-\d+-?\d+?)\.flac/;
+    
+    const match = filePathRegex.exec(filePath);
+    if (!match) {
+        console.log("Wrong Path -- File will be deleted.");
+        bucket.file(object.name).delete();
+        return -1;
+    }
+
+    const uploadCode = match[1], 
+            timestamp = match[2],
+            list = match[match.length - 1], 
+            taskId = match[match.length - 2];
+    
+
+    let supposedAssignee = (await db.ref(`/users`).orderByChild('uploadCode').equalTo(uploadCode).once('value')).val();
+    
+
+    
+    if (!supposedAssignee.exists()) { 
+        console.log("Wrong Upload code -- File will be deleted.");
+        bucket.file(object.name).delete();
+        return -1;
+    }
+
+    // 1. Send a notification to the coordinator
+    const taskRef = await db.ref(`/sound-editing/tasks/${list}/${taskId}`).once('value');
+
+    if (!taskRef.exists()) {
+        console.log("Task does not exist -- File will be deleted.");
+        bucket.file(object.name).delete();
+        return -1;
+    }
+
+    const task = taskRef.val();
+
+    if (['Revise', 'Spare'].indexOf(task.restoration.status) < 0) {
+        console.log("Incorrect task status (only [Revise OR Spare] is allowed here) -- File will be deleted.");
+        bucket.file(object.name).delete();
+        return -1;
+    }
+
+    const coordinator = functions.config().coordinator; // TO
+    const replyTo = task.restoration.assignee.emailAddress;
```

Comment:
Once I had already commented on this — don't use one-time variables, just put this expression directly into the push below.